airflow-ctl/src/airflowctl/api/operations.py

```diff
@@ -243,8 +243,11 @@ def create_event(
     ) -> AssetEventResponse | ServerResponseError:
         """Create an asset event."""
         try:
+            # Ensure extra is an object; DB/UI choke on null
+            if getattr(asset_event_body, "extra", None) is None:
```

Comment:
```suggestion
            if asset_event_body.extra is None:
```

Even though I like `getattr` like safe calls, this actually brings danger to not fail the unit test. If `extra` is removed or changed name is changed in the datamodel, and creates a bug. Because this statement will return None in that case and we always add extra even though the API already changed it. This is why we use Pydantic which is kind of a data contract between the API and the client 

You can try the behavior even with a dict 
```python
a={}
if getattr(a, "extra", None) is None:
    a["extra"] = {}
    print(a)
```

Reply:
Updated as suggested in **aeef5f7** — switched to a direct `asset_event_body.extra is None` check so we don’t mask schema drift; kept the absolute `"/assets/events"` path. Verified locally. Ready for re-review @bugraoz93  thanks!